core/CacheId.php

```diff
@@ -26,4 +26,105 @@ public static function pluginAware($cacheId)
 
         return $cacheId;
     }
+
+    public static function siteAware($cacheId, array $idSites = null)
+    {
+        if ($idSites === null) {
+            $idSites = self::getIdSiteList('idSite');
+            $cacheId .= self::idSiteListCacheKey($idSites);
+
+            $idSites = self::getIdSiteList('idSites');
+            $cacheId .= self::idSiteListCacheKey($idSites);
+
+            $idSites = self::getIdSiteList('idsite'); // tracker param
+            $cacheId .= self::idSiteListCacheKey($idSites);
+        } else {
+            $cacheId .= self::idSiteListCacheKey($idSites);
+        }
+
+        return $cacheId;
+    }
+
+    private static function getIdSiteList($queryParamName)
+    {
+        if (empty($_GET[$queryParamName])
+            && empty($_POST[$queryParamName])
+        ) {
+            return [];
+        }
+
+        $idSiteGetParam = empty($_GET[$queryParamName]) ? [] : explode(',', $_GET[$queryParamName]);
+        $idSitePostParam = empty($_POST[$queryParamName]) ? [] : explode(',', $_POST[$queryParamName]);
+
+        $idSiteList = array_merge($idSiteGetParam, $idSitePostParam);
+        $idSiteList = array_map('intval', $idSiteList);
+        $idSiteList = array_unique($idSiteList);
+        sort($idSiteList);
+        return $idSiteList;
+    }
+
+    private static function idSiteListCacheKey($idSites)
+    {
+        if (empty($idSites)) {
+            return '';
+        }
+
+        if (count($idSites) <= 5) {
+            return '-' . implode('_', $idSites); // we keep the cache key readable when possible
+        } else {
+            return '-' . md5(implode('_', $idSites)); // we need to shorten it
+        }
+    }
+
+    /**
+     * Temporarily overwrites the idSite parameter so all code executed by `$callback()`
+     * will use cache keys for that idSite.
+     *
+     * Useful when you need to change the idSite context for a chunk of code. For example,
+     * if we are archiving for more than one site in sequence, we don't want to use
+     * the same caches for both archiving executions.
+     *
+     * @param string|int $idSite
+     * @param callable $callback
+     * @return mixed returns result of $callback
+     */
+    public static function overwriteIdSiteForCache($idSite, $callback)
+    {
+        // temporarily set the idSite query parameter so archiving will end up using
+        // the correct site aware caches
+        $originalGetIdSite = isset($_GET['idSite']) ? $_GET['idSite'] : null;
+        $originalPostIdSite = isset($_POST['idSite']) ? $_POST['idSite'] : null;
+
+        $originalGetIdSites = isset($_GET['idSites']) ? $_GET['idSites'] : null;
+        $originalPostIdSites = isset($_POST['idSites']) ? $_POST['idSites'] : null;
```

Comment:
Can we check if we are in `TrackerMode` and then also update `idsite` (lowercase)?

Reply:
That would end up adding the ID to the cache twice, right? Eg, for idSite=1, idsite=1, the key would be `$key-1-1`, will that be ok?

Reply:
true... now I wonder if we should do something like `if trackermode then idsite else idSite`... so complicated... I think it would be fine to end up like this as usually in tracking there would be only `idsite` set.

Reply:
Ok, i'll set idsite in 3...2...1...